omnibus/files/private-chef-cookbooks/private-chef/libraries/ec_postgres.rb

```diff
@@ -3,11 +3,28 @@ class EcPostgres
   def self.with_connection(node, database = 'template1', opts = {})
     require 'pg'
     postgres = node['private_chef']['postgresql'].merge(opts)
-    connection = ::PGconn.open('user' => postgres['db_superuser'],
-                               'host' => postgres['vip'],
-                               'password' => postgres['db_superuser_password'],
-                               'port' => postgres['port'],
-                               'dbname' => database)
+    connection = nil
+    retries = 5
+    begin
+      connection = ::PGconn.open('user' => postgres['db_superuser'],
+                                 'host' => postgres['vip'],
+                                 'password' => postgres['db_superuser_password'],
+                                 'port' => postgres['port'],
+                                 'dbname' => database)
+    rescue e
```

Comment:
A quick skim of the PGconn [docs](http://phrogz.net/rubylibs/rdoc/classes/PGconn.html#M000207) didn't turn up anything incredibly useful:

```
PGconn.connect( pghost, pgport, pgoptions, pgtty, dbname, login, passwd ) => conn
PGconn.new( pghost, pgport, pgoptions, pgtty, dbname, login, passwd ) => conn
PGconn.setdb( pghost, pgport, pgoptions, pgtty, dbname, login, passwd ) => conn
PGconn.setdblogin( pghost, pgport, pgoptions, pgtty, dbname, login, passwd ) => conn
pghost:	server hostname (String)
pgport:	server port number (Integer)
pgoptions:	backend options (String)
pgtty:	tty to print backend debug message (ignored in newer versions of PostgreSQL) (String)
dbname:	connecting database name (String)
login:	login user name (String)
passwd:	login password (String)

On failure, it raises a PGError exception.
```

Clicking the link on PGError takes you to a blank page.